src/frontends/tensorflow_common/src/op/pack.cpp

```diff
@@ -16,20 +17,31 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_pack_op(const NodeContext& node) {
-    default_op_checks(node, 1, {"Pack", "PACK"});
-    auto num_size = static_cast<int>(node.get_input_size());
+    default_op_checks(node, 1, {"Pack", "PACK"}, true);
 
     auto axis = node.get_attribute<int64_t>("axis", 0);
+
+    auto num_size = static_cast<int>(node.get_input_size());
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(node.get_input(0).get_node_shared_ptr());
     auto axis_const = make_shared<v0::Constant>(element::i64, Shape{}, axis);
 
+    if (complex_type_mark) {
+        axis_const = make_shared<v0::Constant>(element::i64, Shape{}, axis + 1);
+    } 
```

Comment:
why should we increment `axis` in complex type case? Auxiliary dimension for concatenation of real and imag parts is the last one.
Also, please think how to handle if `axis` is non-negative and negative one.

Now your implementation looks not correct for both non-negative and negatibe cases. Please run your layer tests locally and add negative axis test case.